Framework/Indexing/src/RoundRobinPartitioner.cpp

```diff
@@ -0,0 +1,13 @@
+#include "MantidIndexing/RoundRobinPartitioner.h"
+
+namespace Mantid {
+namespace Indexing {
+
+PartitionIndex
+RoundRobinPartitioner::doIndexOf(const GlobalSpectrumIndex index) const {
+  return PartitionIndex(static_cast<int>(static_cast<size_t>(index) %
+                                         numberOfNonMonitorPartitions()));
+}
```

Comment:
`doIndexOf` is not called or monitors in that case? So monitors are on a dedicated partition, whereas it is round-robin allocation for everything else.